packages/help-center/src/components/help-center-search-results.tsx

```diff
@@ -131,15 +131,20 @@ function HelpSearchResults( {
 		filterManagePurchaseLink( hasPurchases, isPurchasesSection )
 	);
 
-	const routeToQueryMapping = useContextBasedSearchMapping( currentRoute );
+	const { contextSearch, tailoredArticles } = useContextBasedSearchMapping( currentRoute );
 
 	const [ debouncedQuery ] = useDebounce( searchQuery || '', 500 );
 
 	const { data: searchData, isLoading: isSearching } = useHelpSearchQuery(
-		debouncedQuery || routeToQueryMapping,
+		debouncedQuery || contextSearch, // If there's a query, we don't context search
 		locale,
 		{},
-		sectionName
+		sectionName,
+		debouncedQuery
+			? undefined // If there's a query, we don't need tailored articles
+			: tailoredArticles?.find(
+					( tailoredArticle: TailoredArticles ) => tailoredArticle.locale === locale
+			  )
```

Comment:
What do you think of passing the locale to `useContextBasedSearchMapping`, so we can directly return the tailored articles that we need. Then we wouldn't need to have to do the `find` here and the the `find` when we send the `calypso_inlinehelp_article_select` event.

Reply:
Good one, thanks